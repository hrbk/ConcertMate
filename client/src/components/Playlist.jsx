import React from 'react';
import axios from 'axios';
import {Panel} from 'react-bootstrap'
import { connect } from 'react-redux';

const Playlist = (props) => {
	let iframeBaseURL = 'https://open.spotify.com/embed?uri=spotify:';
	let iframeQuery = props.artistId ? 'artist:' + props.artistId + '&theme=black' : 'user:1211115253:playlist:6r1hzKf8a0KYkbt44we1Cl';
	let iframeSRC = iframeBaseURL + iframeQuery;

  return (
  	<div>
	    <Panel collapsible defaultExpanded header="Preview Artist" bsStyle="success">
	    	<iframe src={iframeSRC}
					width="100%" height="95"
					frameBorder="0" allowTransparency="true"></iframe>

	    </Panel>
    </div>
  )
};


const mapStateToProps = (state) => {
  return {
    'artistId': state.artistId //allows this.props.artistId to exist and be accessible in component
  }
}

export const UnwrappedPlaylist = Playlist;
export default connect(mapStateToProps)(Playlist);
