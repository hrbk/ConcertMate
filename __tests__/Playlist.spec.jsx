import React from 'react';
import renderer from 'react-test-renderer';
import Playlist, { UnwrappedPlaylist } from '../client/src/components/Playlist.jsx';

describe('<Playlist />', () => {
  it('should render a Playlist componenet', () => {
    const component = renderer.create(<UnwrappedPlaylist />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
