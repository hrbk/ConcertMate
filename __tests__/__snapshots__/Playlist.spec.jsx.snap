// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playlist /> should render a Playlist componenet 1`] = `
<div>
  <div
    className="panel panel-success"
    id={null}
  >
    <div
      className="panel-heading"
    >
      <h4
        className="panel-title"
        role="presentation"
      >
        <a
          aria-controls={undefined}
          aria-expanded={true}
          aria-selected={true}
          className={null}
          href={undefined}
          onClick={[Function]}
          role={undefined}
        >
          Preview Artist
        </a>
      </h4>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={false}
      className="panel-collapse collapse in"
      id={undefined}
      role={undefined}
    >
      <div
        className="panel-body"
      >
        <iframe
          allowTransparency="true"
          frameBorder="0"
          height="95"
          src="https://open.spotify.com/embed?uri=spotify:user:1211115253:playlist:6r1hzKf8a0KYkbt44we1Cl"
          width="100%"
        />
      </div>
    </div>
  </div>
</div>
`;
